index.js

```diff
@@ -106,6 +106,43 @@ var Controllers = {
     };
   },
 
+  TabBarControllerIOS: function (id) {
+    return {
+      setHidden: function (params) {
+        return RCCManager.TabBarControllerIOS(id, "setTabBarHidden", params);
+      }
+    };
+  },
+
+  Modal: {
+    showLightBox: function(componentId, style) {
+      if (style.backgroundColor !== undefined) {
+        style.backgroundColor = processColor(style.backgroundColor);
+      }
+      RCCManager.modalShowLightBox(componentId, style);
+    },
+    dismissLightBox: function() {
+      RCCManager.modalDismissLightBox();
+    },
+    showController: function(appKey, animated) {
+      var controller = _controllerRegistry[appKey];
+      if (controller === undefined) return;
+      var layout = controller.render();
+      RCCManager.showController(layout, animated);
+    },
+    dismissController: function(animated) {
+      RCCManager.dismissController(animated);
+    }
+  },
+
+  Root: {
```

Comment:
I don't want to add a Root object. Let's keep the root API backwards compatible:
`ControllerRegistry.setRootController(compnentID)`

Let's change the existing `ControllerRegistry.setRootController(compnentID)` to:
`setRootController: function (appKey, animationType = 'none')`
and merge the 2 behaviors, so not passing animationType will keep the existing behavior and passing will have the new behavior. We should also merge the code in the native part so we only have one native setRootController implementation